### PR TITLE
Refactor of i18n conversion scripts (#37)

### PR DESCRIPTION
**Pull request by Jag-Marcel**
### Description
In this PR, I refactored i18n's Scripts directory to account for different versions of Scribe.
- iOS and Android scripts now have separate subdirectories in the Scripts directory
- Scripts and workflows have been edited to account for the restructure
- JSON conversion workflow has been edited to use the `paths:` filter instead of a conditional check job
- Scripts for converting between JSON and xcstrings have been refactored to use Python's `json` module

### Related issues
- #37

### CODE REVIEW

File: .github/workflows/json_conversion.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: json_conversion.yml
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'Scribe-i18n/**.json'
```
Comment (andrewtavis) [inline .github/workflows/json_conversion.yml:8]: Nice! Let's hope this works :) Thanks for the suggestion, @wkyoshida!

File: Scribe-i18n/Scripts/Android/convert_strings_to_json.py [hunk 2]
```diff
@@ -33,8 +32,8 @@ def unescape_special_characters(string):
     matches = regex.findall(content)
     result = dict(matches)
     result = {key: unescape_special_characters(value) for key, value in result.items()}
-    with open(os.path.join(directory,f'{lang}.json'), 'w',encoding='utf-8') as file:
-        json.dump(result, file, indent=4,ensure_ascii=False)
+    with open(os.path.join(directory, f'{lang}.json'), 'w', encoding='utf-8') as file:
+        json.dump(result, file, indent=2, ensure_ascii=False)
```
Comment (andrewtavis) [inline Scribe-i18n/Scripts/Android/convert_strings_to_json.py:36]: Good catch, @Jag-Marcel!

**Comment by andrewtavis** [comment]
One thing for this, @Jag-Marcel, can we get it so that there's a new line at the end of all JSON files. With pre-commit in iOS and Android, this will be enforced on the JSON files and then we'll get conflicts in the subtrees. Aside from this it's looking really good :)

**Comment by Jag-Marcel** [comment]
Here you go, that should fix it

**Review by andrewtavis — APPROVED** [review]
Thanks for all of this and for the quick reaction to the new line request, @Jag-Marcel! 😊 Here's hoping all works out with the workflow and we're good to just start using Scribe-i18n :)

**Pull request merged**